### PR TITLE
Performance Optimisations

### PR DESCRIPTION
**Pull request by keegancsmith**
This increases the performance of `Scorer` via reducing the number of allocations it does to none. Also included is a fix for a regression introduced in #2 where the start of word bonus was awarded for consecutive matches as well.

```
$ go test -run '^$' -bench '.*' -benchmem
Before
BenchmarkScoreASCII-4            2000000               757 ns/op             448 B/op          2 allocs/op

After
BenchmarkScoreASCII-4            3000000               450 ns/op               0 B/op          0 allocs/op
```

### CODE REVIEW

**Comment by keegancsmith** [comment]
cc @slimsag

File: stringscore_test.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func TestScore(t *testing.T) {
 
 func TestExactMatchIsPrefferedOverFuzzyMatch(t *testing.T) {
 	query := "backend"
-	fuzzyScore := stringscore.Score("vendor/github.com/coreos/go-oidc/key/manager.go", query)
+	fuzzyScore := stringscore.Score("vendor/github.com/gorilla/websocket/conn_read_legacy.go", query)
```
Comment (felixfbecker) [inline stringscore_test.go]: why this change?

Comment (keegancsmith) [reply stringscore_test.go]: backend doesn't match it (ie score is 0). So wasn't actually testing what it said it was. Instead I took the longest fuzzy match I could find in our backend codebase.

**Comment by slimsag** [comment]
This is awesome :) thanks for the fix on my regression -- sorry about that!
